### DISCUSSION TITLE
Can't compile http-server with mingw64 for the armv3l

### DISCUSSION THREAD

**Discussion by KGB1st** [cesanta/mongoose#2751] [Ask questions, get answers] [answered]
Compile successfully:
`arm-none-eabi-gcc -D MG_ARCH=MG_ARCH_NEWLIB -c main.c mongoose.c`

>$ file main.o mongoose.o
main.o:     ELF 32-bit LSB relocatable, ARM, EABI5 version 1 (SYSV), not stripped
mongoose.o: ELF 32-bit LSB relocatable, ARM, EABI5 version 1 (SYSV), not stripped

And then got errors while trying to call:
`arm-none-eabi-ld -o main mongoose.o main.o`

```
C:\msys64\mingw64\bin\arm-none-eabi-ld.exe: warning: cannot find entry symbol _start; defaulting to 00008000
...
C:\msys64\mingw64\bin\arm-none-eabi-ld.exe: mongoose.o: in function `mg_dns_free':
mongoose.c:(.text+0x938): undefined reference to `free'
C:\msys64\mingw64\bin\arm-none-eabi-ld.exe: mongoose.o: in function `mg_dns_parse_name_depth':
mongoose.c:(.text+0xbd4): undefined reference to `memcpy'
C:\msys64\mingw64\bin\arm-none-eabi-ld.exe: mongoose.o: in function `mg_dns_parse_rr':
mongoose.c:(.text+0xd18): undefined reference to `memset'
C:\msys64\mingw64\bin\arm-none-eabi-ld.exe: mongoose.o: in function `mg_dns_parse':
mongoose.c:(.text+0xf74): undefined reference to `memset'
C:\msys64\mingw64\bin\arm-none-eabi-ld.exe: mongoose.c:(.text+0x116c): undefined reference to `memcpy'
...
C:\msys64\mingw64\bin\arm-none-eabi-ld.exe: mongoose.o: in function `mg_dtoa':
mongoose.c:(.text+0x2024): undefined reference to `__aeabi_dcmpeq'
C:\msys64\mingw64\bin\arm-none-eabi-ld.exe: mongoose.c:(.text+0x2078): undefined reference to `__aeabi_dcmpgt'
...
C:\msys64\mingw64\bin\arm-none-eabi-ld.exe: mongoose.c:(.text+0x220c): undefined reference to `__aeabi_dcmple'
...
```

```
$ pacman -Q
..
mingw-w64-x86_64-arm-none-eabi-binutils 2.41-2
mingw-w64-x86_64-arm-none-eabi-gcc 12.2.0-1
mingw-w64-x86_64-arm-none-eabi-newlib 4.4.0.20231231-1
mingw-w64-x86_64-gcc-libs 14.1.0-2
....
```

**Comment by scaprile** [comment, marked as answer]
There are tons of examples for arm with gcc. You are missing basic stdlib functions, you are not providing a proper standard library like newlib or equivalent to the linker, something the likes of `--specs `. Please learn your tools, know your system, learn how to build code for your system. We are not currently building for that architecture and compiler and Windows.

That example (all of them, btw) has a Makefile for a reason. Once you solve your current problem, you'll be missing Windows networking functions
https://github.com/cesanta/mongoose/blob/3ae1a0f82ac6c49db0813969fc1b20f0499eaf3b/tutorials/http/http-server/Makefile#L14

You'll quite likely need to indicate the cpu, like https://github.com/cesanta/mongoose/blob/3ae1a0f82ac6c49db0813969fc1b20f0499eaf3b/test/Makefile#L137-L139
